src/par_iter/mod.rs

```diff
@@ -674,11 +671,114 @@ pub trait IndexedParallelIterator: ExactParallelIterator {
     /// iterators are of unequal length, you only get the items they
     /// have in common.
     fn zip<ZIP_OP>(self, zip_op: ZIP_OP) -> ZipIter<Self, ZIP_OP::Iter>
-        where ZIP_OP: IntoParallelIterator, ZIP_OP::Iter: IndexedParallelIterator
+        where ZIP_OP: IntoParallelIterator,
+              ZIP_OP::Iter: IndexedParallelIterator
     {
         ZipIter::new(self, zip_op.into_par_iter())
     }
 
+    /// Lexicographically compares the elements of this `ParallelIterator` with those of
+    /// another.
+    fn cmp<I>(self, other: I) -> Ordering
+        where I: IntoParallelIterator<Item = Self::Item>,
+              I::Iter: IndexedParallelIterator,
+              Self::Item: Ord
+    {
+        self.zip(other.into_par_iter())
+            .map(|(x, y)| Ord::cmp(&x, &y))
+            .reduce(|| Ordering::Equal,
+                    |cmp_a, cmp_b| if cmp_a != Ordering::Equal {
+                        cmp_b
```

Comment:
I think this is backwards. You want `if cmp_a == Ordering::Equal { cmp_ b } else { cmp_a }`.